src/Microsoft.Health.Fhir.Shared.Api/Features/Formatters/HttpContextExtensions.cs

```diff
@@ -9,63 +9,48 @@
 using Hl7.Fhir.Rest;
 using Microsoft.AspNetCore.Http;
 using Microsoft.AspNetCore.Http.Features;
-using Microsoft.Extensions.Logging;
 using Microsoft.Health.Fhir.Core.Features;
 
 namespace Microsoft.Health.Fhir.Api.Features.Formatters
 {
     public static class HttpContextExtensions
     {
-        public static SummaryType GetSummaryType(this HttpContext context, ILogger logger)
+        public static SummaryType GetSummaryTypeOrDefault(this HttpContext context)
         {
             var query = context.Request.Query[KnownQueryParameterNames.Summary].FirstOrDefault();
 
             if (!string.IsNullOrWhiteSpace(query) && context.Response.StatusCode == (int)HttpStatusCode.OK)
             {
-                try
+                if (Enum.TryParse<SummaryType>(query, true, out var summary))
```

Comment:
Could this be an 'and' condition on the if statement above?

Reply:
indubitably

---
In reply to: [587686055](https://github.com/microsoft/fhir-server/pull/1699#discussion_r587686055) [](ancestors = 587686055)